framework/src/main/java/org/checkerframework/framework/type/AbstractViewpointAdapter.java

```diff
@@ -408,6 +414,25 @@ private AnnotatedTypeMirror substituteTVars(AnnotatedTypeMirror lhs, AnnotatedTy
             // Base case where actual type argument is extracted
             if (lhs.getKind() == TypeKind.DECLARED) {
                 rhs = getTypeVariableSubstitution((AnnotatedDeclaredType) lhs, atv);
+                // When substituting an annotated type variable use (e.g., fields, return type), we
+                // don't want to replace the type qualifier of it with the qualifier on the
+                // type argument, as specified in
+                // https://checkerframework.org/manual/#type-variable-use. However, method
+                // getTypeVariableSubstitution will replace the annotated type qualifier as well.
+                // We fix up the substitution by checking if the types of the lower and upper bound
+                // are the same. If they are the same: (1) either the type variable use is
+                // annotated, and we need to fix up the
+                // primary annotation of the substituted result (rhs) by replacing it with the
+                // previous annotated type qualifier (from atv); (2) or the use is not annotated,
+                // but
```

Comment:
Can you try to clarify this comment? I'm not sure I follow what this is trying to do. (Also, improve the formatting of the comment.)